### PR TITLE
(feat) Change background color every 1 second for 10 seconds

### PR DESCRIPTION
**Pull request by haomingh**
Merging this PR will change the background color of document body to a random color every 1 second for 10 seconds.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  var options = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 'A', 'B', 'C', 'D', 'E', 'F'];
+  for (var i = 0; i < 6; i++) {
+    var randomIndex = Math.floor(Math.random() * options.length);
+    color += options[randomIndex];
+  }
+  return color;
+}
+
+function changeBackground() {
+  for (var i = 0; i < 10; i++) {
```
Comment (acvietnogi) [inline main.js]: what if the product requirement changes from 10 seconds to 7.77 seconds?

File: main.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+
+function getRandomColor() {
```
Comment (acvietnogi) [inline main.js]: seems like the color can repeat?

File: main.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  var color = '#';
+  var options = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 'A', 'B', 'C', 'D', 'E', 'F'];
+  for (var i = 0; i < 6; i++) {
+    var randomIndex = Math.floor(Math.random() * options.length);
+    color += options[randomIndex];
+  }
+  return color;
```
Comment (acvietnogi) [inline main.js]: What if the product requirement requires a mix of Hexadecimal, RGB, RGBA, HSL, etc.?

File: main.js (hunk 2 above)
Comment (acvietnogi) [inline main.js]: Seems like`getRandomColor` function has two concerns: 1. Choosing a random color and 2. Generating a hex color code.  Wouldnt it be better to have separation of concerns?

File: main.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+}
+
+function changeBackground() {
```
Comment (acvietnogi) [inline main.js]: Everything is in the global scope, is there a cleaner way?

File: main.js (hunk 2 above)
Comment (haomingh) [inline main.js]: Yes, the color can repeat, but the likelihood of that happening should be very low according to combination theory (16 options per slot for 6 different slots).

File: main.js (hunk 3 above)
Comment (haomingh) [inline main.js]: In this case, I would assume that opacity is 100% for all cases, but still ask the PM or designer to confirm if this is what they want.

As far as implementation for the other format options, there are many ways to do it. One way is to pass an argument called `format` to `getRandomColor`. Within `getRandomColor` we could then check to see what the requested format type is (eg. `'hex'`), and fire off a different helper function like `getRandomHex()` to separate concerns.

File: main.js (hunk 4 above)
Comment (haomingh) [inline main.js]: Yep! We can just wrap everything in a function and export it.

File: main.js (hunk 1 above)
Comment (haomingh) [inline main.js]: Then the background would change 8 times if we changed the 10 -> 7.77

To make the function more robust, we could accept an input argument to tell the function when to stop. Likewise, if we wanted to interval from 1 sec to something else, we could accept an argument for that as well.

**Comment by haomingh** [comment]
Hey Alex,

Thanks for the comments! I just responded to them. Please let me know which items you'd like me to implement in code and I'll be happy to do it!

File: main.js (hunk 1 above)
Comment (acvietnogi) [inline main.js]: I dont believe your suggestion will fix the problem if we only want the colors changing for 7.77 seconds.

File: main.js (hunk 2 above)
Comment (acvietnogi) [inline main.js]: Yea, you're right that it will unlikely happen

File: main.js (hunk 3 above)
Comment (acvietnogi) [inline main.js]: ok, lets keep it as hex

**Comment by haomingh** [comment]
Merging :)

**Pull request merged**